config/matrix-api-v1.yml

```diff
@@ -0,0 +1,299 @@
+  /filters/{filter_name}:
+    get:
+      summary: 'Get a description of a filter'
+      description: >
+        Get detailed information about a filter that can be applied
+        to the HCA expression data.
+      parameters:
+        - name: filter_name
+          schema:
+            type: string
+          required: True
+          in: path
+          description: 'Filter to describe.'
+      responses:
+        '200':
+          description: "Filter found."
+          content:
+            application/json:
+              schema:
+                oneOf:
+                  - $ref: '#/components/schemas/NumericFieldDetail'
+                  - $ref: '#/components/schemas/CategoricalFieldDetail'
+        '404':
+          description: 'Filter not found.'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/MatrixErrorResponse'
+  /fields:
+    get:
+      summary: Get a lsit of available metadata fields
+      description: >
+        Get a list of metadata field names that a POST to 
+        /matrix will accept.
+      responses:
+        '200':
+          description: 'Successfully return field list.'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/MatrixFields'
+  /fields/{field_name}:
+    get:
+      summary: 'Get a description of a metadata field'
+      description: >
+        Get detailed information about a an available metadata field.
+      parameters:
+        - name: field_name
+          schema:
+            type: string
+          required: True
+          in: path
+          description: 'Field to describe.'
+      responses:
+        '200':
+          description: "Field found."
+          content:
+            application/json:
+              schema:
+                oneOf:
+                  - $ref: '#/components/schemas/NumericFieldDetail'
+                  - $ref: '#/components/schemas/CategoricalFieldDetail'
+        '404':
+          description: 'Field not found.'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/MatrixErrorResponse'
+components:
+  schemas:
+    FilterOperation:
+      description: 'Comparison operations allowed in matrix filters.'
+      type: string
+      enum:
+        - '='
+        - '!='
+        - '>'
+        - '<'
+        - 'in'
+    MatrixFormat:
+      description: 'Format for the output matrix.'
+      type: string
+      default: 'loom'
+    MatrixFields:
+      description: 'Metadata fields to include in output matrix.'
+      type: array
+      items:
+        type: string
+    MatrixFilters:
+      description: 'Accepted filter names for a matrix request.'
+      type: array
+      items:
+        type: string
+    MatrixFeature:
+      description: >
+        Feature type to include in the output matrix. For example, 
+        genes vs. transcripts.
+      type: string
```

Comment:
Could this be an enumeration? What additional _feature types_ would we expect in the future?

Reply:
It could be. Right now it's just `gene` or `transcript`, but I think you could have `protein` or some non-genic annotation of a region.